src/Microsoft.AspNet.Mvc.ViewFeatures/Rendering/TagBuilder.cs

```diff
@@ -252,6 +251,15 @@ public void MergeAttribute(string key, string value, bool replaceExisting)
         /// <inheritdoc />
         public void WriteTo(TextWriter writer, HtmlEncoder encoder)
         {
+            var htmlTextWriter = writer as HtmlTextWriter;
+            if (htmlTextWriter != null)
+            {
+                // As a perf optimization, we can buffer this output rather than writing it
+                // out character by character.
+                htmlTextWriter.Write(this);
+                return;
```

Comment:
It would be great to see some impact for these kind of changes. What's the effect of this?